server/api.go

```diff
@@ -690,3 +700,101 @@ func (p *Plugin) updateSettings(w http.ResponseWriter, r *http.Request) {
 	resp, _ := json.Marshal(info.Settings)
 	w.Write(resp)
 }
+
+func (p *Plugin) getRepositories(w http.ResponseWriter, r *http.Request) {
+	userID := r.Header.Get("Mattermost-User-ID")
+	if userID == "" {
+		http.Error(w, "Not authorized", http.StatusUnauthorized)
+		return
+	}
+
+	ctx := context.Background()
+
+	var githubClient *github.Client
+
+	if info, err := p.getGitHubUserInfo(userID); err != nil {
+		writeAPIError(w, err)
+		return
+	} else {
+		githubClient = p.githubConnect(*info.Token)
+	}
+
+	repositories, _, err := githubClient.Repositories.List(ctx, "", &github.RepositoryListOptions{})
+	if err != nil {
+		mlog.Error(err.Error())
+	}
+
+	resp, _ := json.Marshal(repositories)
+	w.Write(resp)
+}
+
+func (p *Plugin) createIssue(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodPost {
+		http.Error(w, fmt.Sprintf("Request: %s is not allowed, must be POST", r.Method), http.StatusMethodNotAllowed)
+		return
+	}
+
+	userID := r.Header.Get("Mattermost-User-ID")
+	if userID == "" {
+		http.Error(w, "Not authorized", http.StatusUnauthorized)
+		return
+	}
+
+	// Make sure user have a connected github account
+	var githubClient *github.Client
+	info, err := p.getGitHubUserInfo(userID)
+	if err != nil {
+		writeAPIError(w, err)
+		return
+	}
+
+	githubClient = p.githubConnect(*info.Token)
+
+	// get data for the issue from the request body and fill IssueRequest object
+	issue := &IssueRequest{}
+	dec := json.NewDecoder(r.Body)
+	if err := dec.Decode(&issue); err != nil {
+		mlog.Error("Error decoding JSON body", mlog.Err(err))
+		writeAPIError(w, &APIErrorResponse{ID: "", Message: "Please provide a JSON object.", StatusCode: http.StatusBadRequest})
+		return
+	}
+
+	if issue.Title == "" {
+		writeAPIError(w, &APIErrorResponse{ID: "", Message: "Please provide a valid issue title.", StatusCode: http.StatusBadRequest})
+		return
+	}
+
+	if issue.Body == "" {
```

Comment:
0/5 we should allow posting issues with empty bodies, @aaronrothschild? Also, what do we do in the Jira plugin?

Reply:
Can wait